float32[] matrix
float32 size 
string folder


